File: WoB_Server/src/networking/response/ResponseSeasonChange.java
/**
 * @author Lobby Team
 */
package networking.response;

// Other Imports
import metadata.Constants;
import utility.GamePacket;

/**
 * The ResponseLogin class contains information about the authentication
 * process.
 */
public class ResponseSeasonChange extends GameResponse {

    private short status;
    private int seasonCode;
    private int eventCode;


    public ResponseSeasonChange() {
        responseCode = Constants.SMSG_SEASON_CHANGE;
    }

    @Override
    public byte[] constructResponseInBytes() {
        GamePacket packet = new GamePacket(responseCode);
        packet.addShort16(status);

        if (status == 0) { //season change, call season
            packet.addInt32(seasonCode);            
        }
        else if (status == 1){ //natural event change, call natural event
        	packet.addInt32(eventCode); 
        }
        else if (status == 2){
        	//packet.addString();
        }

        return packet.getBytes();
    }

    public void setStatus(short status) {
        this.status = status;
    }
    
    public void setSeasonCode(int seasonCode) {
        this.seasonCode = seasonCode;
    }
    
    public void setEventCode(int eventCode) {
        this.eventCode = eventCode;
    }
}

File: WoB_Client/Assets/Network/Response/ResponseSeasonChange.cs
//author: Lobby Team

using UnityEngine;

using System;

public class ResponseSeasonChangeEventArgs : ExtendedEventArgs {
	public short status { get; set; }
	public int env_id { get; set; }
	
	public ResponseSeasonChangeEventArgs() {
		event_id = Constants.SMSG_SEASON_CHANGE;
	}
}

public class ResponseSeasonChange: NetworkResponse {
	
	private int event_id;
	private short status;
	
	public ResponseSeasonChange() {
	}
	
	public override void parse() {
		status = DataReader.ReadShort(dataStream);
		event_id = DataReader.ReadInt(dataStream);
	}
	
	public override ExtendedEventArgs process() {
		ResponseSeasonChangeEventArgs args = new ResponseSeasonChangeEventArgs();
		args.env_id = event_id;
		args.status = status;

		return args;
	}
}

